integration/map_test.go

```diff
@@ -119,3 +198,84 @@ func TestInclusionWithEnv(t *testing.T) {
 		}
 	}
 }
+
+func TestInclusionBatch(t *testing.T) {
+	ctx := context.Background()
+	env, err := integration.NewMapEnv(ctx, "TestInclusionBatch")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer env.Close()
+TestCase:
+	for _, tc := range []struct {
+		desc string
+		trillian.HashStrategy
+		batchSize, numBatches int
+	}{
+		{
+			desc:         "maphasher batch",
+			HashStrategy: trillian.HashStrategy_TEST_MAP_HASHER,
+			batchSize:    64,
+			numBatches:   32,
+		},
+	} {
+		tree, hasher, err := newTreeWithHasher(ctx, env, tc.HashStrategy)
+		if err != nil {
+			t.Errorf("%v: newTreeWithHasher(%v): %v", tc.desc, tc.HashStrategy, err)
+		}
+		mapID := tree.TreeId
+		client := env.MapClient
+
+		leafBatch := make([][]*trillian.MapLeaf, tc.numBatches)
+		leafMap := make(map[string]*trillian.MapLeaf)
+		for i := range leafBatch {
+			leafBatch[i] = createMapLeaves(makeBatchIndexValues(i, tc.batchSize)...)
+			for _, l := range leafBatch[i] {
+				leafMap[hex.EncodeToString(l.Index)] = l
+			}
+		}
+
+		for _, b := range leafBatch {
+			if _, err := client.SetLeaves(ctx, &trillian.SetMapLeavesRequest{
+				MapId:  mapID,
+				Leaves: b,
+			}); err != nil {
+				t.Errorf("%v: SetLeaves(): %v", tc.desc, err)
+				continue TestCase
+			}
+		}
+
+		// Shuffle the indexes. Map access is randomized.
+		indexBatch := make([][][]byte, 0, tc.numBatches)
+		i := 0
+		for _, v := range leafMap {
+			if i%tc.batchSize == 0 {
```

Comment:
Something like this is a little less optimal in terms of reallocation, but a lot easier to see what's going on:

```golang
if i%tc.batchSize == 0 {
    indexBatch = append(indexBatch, []byte{})
 }
 indexBatch[i/tc.batchSize] = append(indexBatch[i/tc.batchSize], v.Index)
 i++
...
```